apis_core/generic/api_views.py

```diff
@@ -31,6 +31,26 @@ def get_serializer_class(self):
         serializer_class_modules = module_paths(
             self.model, path="serializers", suffix="Serializer"
         )
+        prefix = makeclassprefix(self.request.accepted_renderer.format)
+        serializer_class_modules = (
+            module_paths(self.model, path="serializers", suffix=f"{prefix}Serializer")
+            + serializer_class_modules
+        )
+
+        if dialect := self.request.query_params.get("dialect"):
```

Comment:
given that `dialect` can take any string and uses the inheritance search it basically silently fails and returns just the basic serialization. I think we should:

- add the dialect as a search param in one of the basic `FilterClasses` and create an enum of allowed values similar to the search through class inheritance chain down below. Like that the allowed/configured dialects would end in the swagger definition
- and raise a proper error when the search down below doesnt find any serializer classes with the dialect name in it. Like that people get what they requestet or they get an error (but not something they didnt expect)

Reply:
Oke, I took a little time to play around with this. If we want to have `dialect` as a parameter with enum values for the API, this basically comes down to adding a `ChoiceFilter` to the filterset class that is used for the API. We can not add this `ChoiceFilter` to the `GenericFilterSet`, because this is also used for the HTML filters in the list view.

We could inject the `ChoiceFilter` into the used filterset in the `GenericFilterBackend` *after* the filterset is chosen. But the users still have to define the choices somewhere. So why not let it be the responsiblity of the users and let them simply create a custom `ApiFilterSet` and add the `ChoiceFilter` themselves?